packages/neon-core/src/u/misc.ts

```diff
@@ -78,3 +79,45 @@ export function reverseHex(hex: string): string {
   }
   return out;
 }
+
+/**
+ * Calculates the byte size of any supported input following NEO's variable int format.
+ */
+interface NeonSerializable {
+  size: number;
+  serialize: () => string;
+}
+type Serializables = number | HexString | NeonSerializable | Serializables[];
+export function getSerializedSize(value: Serializables): number {
+  switch (typeof value) {
+    case "number": {
+      if (value < 0xfd) return 1;
+      else if (value <= 0xffff) return 3;
+      else return 5;
+    }
+    case "object": {
+      if (value instanceof HexString) {
+        const size = value.byteLength;
+        return getSerializedSize(size) + size;
+      } else if (Array.isArray(value)) {
+        const array_length = value.length;
+        let size = 0;
+        if (array_length > 0) {
+          size =
+            value
+              .map((item) => getSerializedSize(item))
+              .reduce((prev, curr) => prev + curr, 0) - array_length;
+        }
+        return getSerializedSize(array_length) + size;
+      } else if (
+        typeof value.size === "number" &&
+        typeof value.serialize === "function"
+      ) {
+        return getSerializedSize(value.size) + value.size;
+        // do not break here so we fall through to the default
+      }

```

Comment:
```suggestion
      }
        // do not break here so we fall through to the default
```
Flip the lines so the comment is clearly meant for outside the if clause.